Framework/API/src/SpectrumInfo.cpp

```diff
@@ -0,0 +1,42 @@
+#include "MantidAPI/GeometryInfoFactory.h"
+#include "MantidAPI/GeometryInfo.h"
+#include "MantidAPI/SpectrumInfo.h"
+#include "MantidKernel/MultiThreaded.h"
+#include "MantidKernel/make_unique.h"
+
+namespace Mantid {
+namespace API {
+
+SpectrumInfo::SpectrumInfo(const MatrixWorkspace &workspace)
+    : m_factory{Kernel::make_unique<GeometryInfoFactory>(workspace)},
```

Comment:
Basing `SpectrumInfo` on `GeometryInfo` was just an intermediate step, such that we could quickly see if the interface and the interaction with `MatrixWorkspace` was ok. I will push some more changes that clean this up in a minute.